### ISSUE TITLE
Push code for Github

### ISSUE THREAD

**Issue by miguelmartins17** [dwyl/apprenticeship#33]
How can I push the code of an entire application?
Do I need to push file to file or folder to folder or can I do it all right?

**Comment by nelsonic** [comment]
@miguelmartins17 good question. if you are in the root of the project simply run `git add .` (_the `.` means "**add everything** in this directory and all sub-directories_). e.g:
```
cd learn-flutter
git add .
git commit -m 'add complete code for TDD & Architecture Video 9 for issue #15'
```

**Comment by miguelmartins17** [comment]
After executing these commands this was the result.

![Screen Shot 2020-03-02 at 10 12 23](https://user-images.githubusercontent.com/27420533/75667957-6f341c00-5c70-11ea-9f6f-ae6932c085e7.png)

**Comment by nelsonic** [comment]
@miguelmartins17 that's strange. 😕 
Did you add the project files to the `learn-flutter/tdd-architecture-example` directory? 💭 
If you want to show us your desktop via Zoom call, just message in `Signal/@dwyl` 💻

**Comment by iteles** [comment]
It might also help to run the `git status` command and `git log` to get a fuller picture.

**Comment by miguelmartins17** [comment]
This is the result when I run the `git log` command.

![Screen Shot 2020-03-02 at 16 51 48](https://user-images.githubusercontent.com/27420533/75698222-3b73e900-5ca6-11ea-89a9-58fa091d2b7a.png)

**Comment by nelsonic** [comment]
@miguelmartins17 after our Zoom call do you feel more confident in your `git` skills?

**Comment by miguelmartins17** [comment]
@nelsonic I tried to push the code to the GitHub but when I run the `git push` command this shows up.

![Screen Shot 2020-03-09 at 10 54 29](https://user-images.githubusercontent.com/27420533/76210233-9d2ed880-61fb-11ea-81f5-17f08f46a1ea.png)

**Comment by nelsonic** [comment]
@miguelmartins17 what did Google say when you pasted the error message and searched? 💭

**Comment by nelsonic** [comment]
The solution is right there in the yellow text. You need to run `git pull` _before_ `git push` because there are changes on GitHub that are not reflected on your `localhost` ...

**Comment by miguelmartins17** [comment]
@nelsonic I tried to look for answers like using git pull but this only comes to me when I run the command.

![Screen Shot 2020-03-09 at 22 02 47](https://user-images.githubusercontent.com/27420533/76261385-c5e1bd00-6251-11ea-8dec-f1fdedafa8dd.png)

**Comment by nelsonic** [comment]
@miguelmartins17 which branch do you _want_ to be working on?
(_maybe it's time that you give your branches more descriptive names_) 💭

**Comment by miguelmartins17** [comment]
The branch `miguelmartins17-patch-4` didn't show up at the terminal so I created it. Of course I should give the branches a better name.

**Comment by miguelmartins17** [comment]
@nelsonic I think I'm forgetting some steps so I can put the project in the GitHub.

**Comment by nelsonic** [comment]
@miguelmartins17 agreed. Please revisit the Git guide: https://git-scm.com/book/en/v2
The better you know Git the faster you can get your work done.
We all take Git skills for granted much like we do with arithmetic, but it's worth knowing well.

**Comment by miguelmartins17** [comment]
I ran the `git push` command and this appears.

![Screen Shot 2020-03-12 at 13 09 46](https://user-images.githubusercontent.com/27420533/76525151-3c093e00-6463-11ea-90f8-6fcd86dae7ab.png)

So I tried to run the command using the way they describe it there:
`git push --set-upstream origin miguelmartins17-patch-4`.


![Screen Shot 2020-03-12 at 13 09 57](https://user-images.githubusercontent.com/27420533/76525265-6eb33680-6463-11ea-94a2-7c094d237776.png)

After that I thought about using `git pull`, and I got this result.

![Screen Shot 2020-03-09 at 22 02 47](https://user-images.githubusercontent.com/27420533/76525387-a8843d00-6463-11ea-98f7-937c5eff9e9f.png)

**Comment by nelsonic** [comment]
@miguelmartins17 did you run the suggested command?
```sh
git branch --set-upstream-to=miguelmartins17-patch4
```
Followed by:

```sh
git pull
```

**Comment by nelsonic** [comment]
@miguelmartins17 any luck/update?

**Comment by nelsonic** [comment]
@miguelmartins17 please run this commend in your terminal:
```
git remote set-url origin git@github.com:dwyl/flutter-counter-example.git
git push -u origin master
```

**Comment by miguelmartins17** [comment]
https://github.com/dwyl/flutter-counter-example

I finished on pushing the empty project to this repository.

**Comment by nelsonic** [comment]
@miguelmartins17 now that you have your SSH key setup for GitHub, 🔐 
you should be able to push code to GitHub without any issues. 👍

**Comment by miguelmartins17** [comment]
@nelsonic Thanks!!! 😄
